memgpt/schemas/message.py

```diff
@@ -120,18 +121,45 @@ def dict_to_message(
 
             # Convert from 'function' response to a 'tool' response
             # NOTE: this does not conventionally include a tool_call_id, it's on the caster to provide it
-            return Message(
+            message_args = dict(
                 user_id=user_id,
                 agent_id=agent_id,
                 model=model,
                 # standard fields expected in an OpenAI ChatCompletion message object
-                role="tool",  # NOTE
+                role=MessageRole.tool,  # NOTE
                 text=openai_message_dict["content"],
                 name=openai_message_dict["name"] if "name" in openai_message_dict else None,
                 tool_calls=openai_message_dict["tool_calls"] if "tool_calls" in openai_message_dict else None,
                 tool_call_id=openai_message_dict["tool_call_id"] if "tool_call_id" in openai_message_dict else None,
                 created_at=created_at,
             )
+            if id is not None:
```

Comment:
@sarahwooders I couldn't figure out a clean way to refactor this such that when `id` (the kwarg) is `not None`, we pass it through, and when it is `None`, we omit it (and let `Message`'s constructor do the default init).

I tried making a `message_args = dict(...)` version where we then add `id` to the arg dictionary, then do `Message(**message_args)`, but that started throwing Pydantic validation errors, so I just did the simple code duplication version.